assets/js/admin-script.js

```diff
@@ -585,4 +585,34 @@ jQuery( function( $ ) {
 
 	//----------> /Preview <----------//
 
+	function settingsAccordion() {
+		// Hide all the content initially except the "init"
+		$( '.settings_category' ).not( '#init' ).find( '.form-table' ).hide();
+
+		let state_exists = false;
+		// Retrieve the state from localStorage
+		$( '.settings_category h2' ).each( function( index ) {
+			const state = localStorage.getItem( 'wcpdf_accordion_state' + index );
+			if ( 'true' === state ) {
+				state_exists = true;
+				$( this ).addClass( 'active' ).next( '.form-table' ).show();
+			}
+		} );
+
+		// Show the first accordion by default if there's no state saved
+		if ( ! state_exists ) {
+			$( '.settings_category h2:first' ).addClass( 'active' ).next( '.form-table' ).show();
+		}
+
+		$('.settings_category h2' ).click( function() {
+			const index = $( '.settings_category h2' ).index( $( this ) );
+
+			$( this ).toggleClass( 'active' ).next( '.form-table' ).slideToggle( 'fast', function() {
```

Comment:
@Terminator-Barbapapa For the full-screen mode, where the `display: table;`, I think there is no easy way to fix it.

One way would be to make the `display` property `block`, then make the transition, and then revert it to `table`, which has other side effects and issues, such as the bad display of elements during the transition.

Another option would be to switch to height transition instead of toggling the display.

Another option would be wrapping the `form-table` around a DIV element and working with that, but this is not a good way as it requires many modifications.

I tried to fix it using the first option, but several issues appeared that made me think this was not a good approach.

If we're going to fix this issue for full-screen mode, then switching to height transition might be the better option here.

Would you happen to have any suggestions or opinions?

Reply:
I think we can achieve the same animation without `slideToggle()` by manipulating the `max-height` property and set a `transition` in CSS.